### PR TITLE
JOSS paper

### PR DESCRIPTION
**Pull request by naik-aakash**
- [x] Finalize associated text
- [x] Cite other works using this package

### CODE REVIEW

**Comment by naik-aakash** [comment]
Hi @ajjackson , @QuantumChemist and @kaueltzen , It would be great if you can have a look at the [documentation](https://jageo.github.io/LobsterPy/index.html) and the [draft](https://github.com/JaGeo/LobsterPy/files/13612886/JOSS_LobsterPy_v1.pdf) prepared for the submission to JOSS.

@ajjackson: We know you are on parental leave, so do it at your convenience.

I am looking forward to your comments, suggestions and corrections if you spot any erros 😄

**Comment by naik-aakash** [comment]
Thanks @kaueltzen  for the suggestion . I will make the changes :smiley:

**Comment by QuantumChemist** [comment]
Hey! I agree with Katharina and I also found one typo on **Getting started** (LOBTSER), as well as **Fundamental Aspects**, I'd rather say "summed COHPs" instead of "summed cohps". My paper suggestions go directly to @naik-aakash as well :)

**Comment by naik-aakash** [comment]
> Hey! I agree with Katharina and I also found one typo on **Getting started** (LOBTSER), as well as **Fundamental Aspects**, I'd rather say "summed COHPs" instead of "summed cohps". My paper suggestions go directly to @naik-aakash as well :)

Hey thanks for spotting the mistake as usual. 😄 . Will fix it

**Comment by naik-aakash** [comment]
Hi @kaueltzen and @QuantumChemist, I re-read the fundamentals section. But in steps 1 and 2, it is made clear the COHP data is explicitly for the bonds at the site. I think it should be clear to the users from this that COHP data is for the bonds at the site.

But if you prefer any specific way to phrase it in the steps, please share exactly what line should be changed or simply raise a PR.

**Comment by naik-aakash** [comment]
[Revised draft](https://github.com/naik-aakash/LobsterPy/suites/19035839669/artifacts/1113967930)

**Comment by naik-aakash** [comment]
> Hi, I sent my suggestions for the paper draft directly to @naik-aakash .
> 
> Regarding the documentation: it's very detailed and well structured, just one small thing: in https://jageo.github.io/LobsterPy/fundamentals/index.html the first COHP plot is referenced as _COHP data for a site_ and later on it says _Similarly, the antibonding percentage is evaluated for the site from the COHP data._ etc. Although the data is stored site-wise, I would clarify here that the plot contains "COHP data for a site and a specific neighbor / bonding partner", i.e. something "bond-wise", not "site-wise". Otherwise it could confuse new users with no prior experience in bonding analysis.

Thanks ! Has been addressed now as part of #208

**Comment by naik-aakash** [comment]
> Hi @kaueltzen and @QuantumChemist, I re-read the fundamentals section. But in steps 1 and 2, it is made clear the COHP data is explicitly for the bonds at the site. I think it should be clear to the users from this that COHP data is for the bonds at the site.
> 
> But if you prefer any specific way to phrase it in the steps, please share exactly what line should be changed or simply raise a PR.

**Review by JaGeo — CHANGES_REQUESTED** [review]
I made some comments. I hope it helps!

File: paper/paper.md
Comment (JaGeo) [reply paper/paper.md:51]: You could add a typical reference or book here? Hoffmann or one of the other famous books on bonds in solids!

Comment (JaGeo) [reply paper/paper.md:51]: You could even add Drons' latest book as well

File: paper/paper.md
Comment (naik-aakash) [reply paper/paper.md:55]: Oh yes, Will add it 😃

File: paper/paper.md
Comment (naik-aakash) [reply paper/paper.md:36]: Yes, indeed !

**Comment by JaGeo** [comment]
Hi @ajjackson ,

We are thinking about submitting the JOSS in one of the next weeks. Would you have time to agree on the paper draft and submission in the next two weeks?

JG

File: paper/paper.md
Comment (naik-aakash) [reply paper/paper.md:57]: You mean the abbreviation or in general like how to use it for ML studies ? 

I ended up removing the full form when editing text, will add it at the beginning

Comment (JaGeo) [reply paper/paper.md:57]: Make sure you introduce the abbreviation. 😀

**Comment by naik-aakash** [comment]
[PDF draft](https://github.com/naik-aakash/LobsterPy/actions/runs/7385519369/artifacts/1142943919)

File: paper/paper.md
Comment (naik-aakash) [reply paper/paper.md]: Done

**Comment by JaGeo** [comment]
@naik-aakash , maybe you can add a new draft and then we wait for the approval of all co-authors!

**Comment by naik-aakash** [comment]
> @naik-aakash , maybe you can add a new draft and then we wait for the approval of all co-authors!

Ahh, I had just update the link to previous pdf draft.

Putting it here to avoid confusion

[Draft latest](https://github.com/naik-aakash/LobsterPy/actions/runs/7386643689/artifacts/1143139245)

**Comment by JaGeo** [comment]
There is an issue with the first reference, @naik-aakash

**Comment by naik-aakash** [comment]
> There is an issue with the first reference, @naik-aakash

Has been fixed now.

 Link to latest [PDF draft](https://github.com/naik-aakash/LobsterPy/actions/runs/7399140258/artifacts/1145520029)

**Comment by naik-aakash** [comment]
Link to latest [PDF draft](https://github.com/naik-aakash/LobsterPy/actions/runs/7399428894/artifacts/1145587468)

**Comment by naik-aakash** [comment]
Link to latest [PDF draft](https://github.com/naik-aakash/LobsterPy/actions/runs/7399428894/artifacts/1145587468)

**Comment by ajjackson** [comment]
> Hi @ajjackson ,
> 
> We are thinking about submitting the JOSS in one of the next weeks. Would you have time to agree on the paper draft and submission in the next two weeks?
> 
> JG

Hi @JaGeo thanks for the nudge. I am coming to the end of my parental leave now and will able to look at this in the next few days 😄 

-Adam

**Comment by JaGeo** [comment]
@ajjackson , thank you so much!

**Comment by naik-aakash** [comment]
Hi @JaGeo , @ajjackson ,  @QuantumChemist and @kaueltzen , I have just added the DOI in the references which were missing earlier and very small change in text where I cite atomate2, instead of URL.

Here is the updated [draft PDF](https://github.com/naik-aakash/LobsterPy/actions/runs/7445006771/artifacts/1153459679)  

# Todo

- [ ] Before submission to JOSS, need to remove the paper compiling workflow from this PR

**Comment by ajjackson** [comment]
Hi everyone, thanks for moving on with this and sorry I haven't been around much lately.

This looks pretty good already :smile:  If time allows I might have a go at twiddling some words around, would the best way to do that be as PR to Aakash's branch?

I think the Statement of Need has the right emphasis: the package supports high-throughput studies (with citation of this application) and is also intended to also support ML. Both of those things require transformation of data into a consistent, machine-friendly form and that's what this package is for. In some ways for JOSS review this is the most important part of the manuscript to get right.

One area worth a quick discussion:
I don't think the current manuscript makes it entirely clear what the scope of this package is within the ecosystem of other packages it works with.

We describe it as a package that "provides convenient tools to systematically analyze, describe and visualize such LOBSTER computations results." It's unclear from this quite what we mean by "analyze" -- particularly, whether:
- lobsterpy implements the "LobsterEnv" spatial assignment (that code is actually in Pymatgen)
- there are some arbitrary/expert choices that lobsterpy makes in its data processing (I think it is just the threshold to select significant bonds?)

A more complete version of my affiliation is: Scientific Computing Department, Science & Technology Facilities Council, Rutherford Appleton Laboratory, Didcot, 0X11 0QX, UK (I think that's the same as the ChemPlusChem paper). If it's important to save space the laboratory and postcode can be dropped but it's important to keep the department name.

**Comment by naik-aakash** [comment]
> Hi everyone, thanks for moving on with this and sorry I haven't been around much lately.
> 
> This looks pretty good already 😄 If time allows I might have a go at twiddling some words around, would the best way to do that be as PR to Aakash's branch?
> 
> I think the Statement of Need has the right emphasis: the package supports high-throughput studies (with citation of this application) and is also intended to also support ML. Both of those things require transformation of data into a consistent, machine-friendly form and that's what this package is for. In some ways for JOSS review this is the most important part of the manuscript to get right.
> 
> One area worth a quick discussion: I don't think the current manuscript makes it entirely clear what the scope of this package is within the ecosystem of other packages it works with.
> 
> We describe it as a package that "provides convenient tools to systematically analyze, describe and visualize such LOBSTER computations results." It's unclear from this quite what we mean by "analyze" -- particularly, whether:
> 
> * lobsterpy implements the "LobsterEnv" spatial assignment (that code is actually in Pymatgen)
> * there are some arbitrary/expert choices that lobsterpy makes in its data processing (I think it is just the threshold to select significant bonds?)
> 
> A more complete version of my affiliation is: Scientific Computing Department, Science & Technology Facilities Council, Rutherford Appleton Laboratory, Didcot, 0X11 0QX, UK (I think that's the same as the ChemPlusChem paper). If it's important to save space the laboratory and postcode can be dropped but it's important to keep the department name.

Hi @ajjackson , thanks for checking it out already. I have sent an invite to collaborate in my repo fork, so you can directly add changes to this PR.

Regarding your question about "analyze",  Agreed that LobsterPy uses  `LobsterEnv ` from pymatgen, but through this package we also provide the outputs that are way easier to understand in the form of text description and jsons besides the choices set for the thresholds to select significant bonds. Also means to getting plots automatically which help in interpreting the results to the user.

Also, now we added the option of [get_lobster_calc_quality_summary](https://github.com/JaGeo/LobsterPy/blob/2c5accf82b0baa04332db8f3263c2709ed3ae82c/lobsterpy/cohp/analyze.py#L1432), which in principle reads outputs of calculations to provide a summary of how good the LOBSTER calculation is, which I also considered as an "analyze" aspect.

No idea at the moment how else to phrase it in a better way. If you have any suggestions for this, please let me know, would be happy to make the changes or have a go at twiddling the phrase if you get some time in the next days. 

Either way works fine 😄  

But anyway, thanks again!!

**Comment by JaGeo** [comment]
Just to add here: it might be still worth adding that it works very well together with the Materials Project ecosystem :). I think we should mention this as well! Thanks, @ajjackson .

You can, of course, also make a direct review of the paper, if this is easier.

**Comment by naik-aakash** [comment]
Hi, I have updated the text to better explain where LobsterPy as a package fits in. Hope this helps to clarify the points raised by @ajjackson 

Here is the [Updated PDF](https://github.com/naik-aakash/LobsterPy/actions/runs/7463169589/artifacts/1157060082)

**Comment by QuantumChemist** [comment]
> Hi all , I have updated the text to better explain where LobsterPy as a package fits in. Hope this helps to clarify the points raised by @ajjackson
> 
> Here is the [Updated PDF](https://github.com/naik-aakash/LobsterPy/actions/runs/7463169589/artifacts/1157060082)

I've sent my suggestions directly to Aakash :)

**Comment by kaueltzen** [comment]
> > Hi all , I have updated the text to better explain where LobsterPy as a package fits in. Hope this helps to clarify the points raised by @ajjackson
> > Here is the [Updated PDF](https://github.com/naik-aakash/LobsterPy/actions/runs/7463169589/artifacts/1157060082)
> 
> I've sent my suggestions directly to Aakash :)

Same

**Comment by naik-aakash** [comment]
[Updated PDF](https://github.com/naik-aakash/LobsterPy/actions/runs/7485718314/artifacts/1161701846)

**Comment by naik-aakash** [comment]
[Updated PDF](https://github.com/naik-aakash/LobsterPy/actions/runs/7486168774/artifacts/1161782201)

**Comment by ajjackson** [comment]
I'm happy for this to be submitted, I think those points are made clear now 😄

**Comment by JaGeo** [comment]
Awesome! Thank you so much, @ajjackson !

We will think about reviewers, release a new version and then submit! 😃

**Review by JaGeo — APPROVED** [review]

**Pull request merged**
